app/services/mixtape_generator.rb

```diff
@@ -0,0 +1,20 @@
+class MixtapeGenerator
+  def initialize(genre:)
+    @genre = genre
+  end
+
+  def self.perform(genre:)
+    new(genre: genre).perform
+  end
+
+  def perform
+    Mixtape.new(genre: @genre, songs: pick_songs)
+  end
+
+  def pick_songs
```

Comment:
I'm not sure that there is a benefit (eagerloading) to have both the `includes` and `joins`.

Can you get away with just the `joins` and have bullet still say that there's no N+1 query?

From a chaining perspective, tend to order the chains:

* includes/joins
* where
* order by

So, in this case, I'd put `order_by_rand` last in the chain.

This is b/c you might do `order_by_rand(5)` which says give me 5 random songs that uses a `limit` and that's typically the last on the query.

We try to match the order that the SQL might generate.

So try a `.to_sql` after this below and you'll see the `order` be added last.